app/scripts/models/account.js

```diff
@@ -810,7 +810,10 @@ define(function (require, exports, module) {
           .fail(function (err) {
             // If no oauth token existed, or it has gone stale,
             // get a new one and retry.
-            if (ProfileClient.Errors.is(err, 'UNAUTHORIZED')) {
+            if (ProfileClient.Errors.is(err, 'INVALID_TOKEN')) {
+              // invalid token can happen if user uses 'Disconnect' in Firefox Desktop.
+              self.set('accessToken', null);
```

Comment:
> Can you use unset instead of set to null?

Yeap

> Should we also unset sessionToken and sessionTokenContext?

Yeah I think we can create some kind of a helper for this.